incrementalbuild/src/main/java/io/takari/incrementalbuild/spi/FailOnErrorState.java

```diff
@@ -0,0 +1,12 @@
+package io.takari.incrementalbuild.spi;
+
+/***
+ *  The purpose is to indicate if a MojoException must be throw after the compilation if errors are present
+ *  NONE means no preference when errors are present, this is the default value
+ *  TRUE means stop the build if errors are present
+ *  FALSE means continue the build if errors are present
+ */
+public enum FailOnErrorState {
```

Comment:
This is not useful, please remove. Errors reported through a build context either cause or don't cause build failure, there is no third alternative.

Reply:
this was introduced to avoid issues with default value, but now we have the IlegalStateException, if you're good with that.. we can back it to boolean.